crates/types/src/operator.rs

```diff
@@ -0,0 +1,38 @@
+use std::fmt;
+
+use hex::ToHex;
+use libp2p_identity::secp256k1::PublicKey;
+
+/// OperatorPubKey serves as an identifier of protocol entity.
+/// De facto this is a wrapper over `secp256k1::PublicKey`.
+#[derive(
+    serde::Serialize, serde::Deserialize, Debug, Clone, Eq, PartialEq, Hash, Ord, PartialOrd,
+)]
+pub struct OperatorPubKey(pub Vec<u8>);
```

Comment:
Just making the inner type public is not idiomatic, as the usage of `.0` syntax usually is not great. Better implement `AsRef<[u8]>` and some `into_*` getter